server/player/player.go

```diff
@@ -1584,6 +1585,21 @@ func (p *Player) Position() mgl64.Vec3 {
 	return p.pos.Load().(mgl64.Vec3)
 }
 
+// Velocity returns the players current velocity. If there is an attached session, this will be empty.
+func (p *Player) Velocity() mgl64.Vec3 {
+	return p.vel.Load().(mgl64.Vec3)
+}
+
+// SetVelocity updates the players velocity. If there is an attached session, this will just send
+// the velocity to the player session for the player to update.
+func (p *Player) SetVelocity(velocity mgl64.Vec3) {
+	if p.session() == session.Nop {
+		p.vel.Store(velocity)
+		return
+	}
+	p.session().SendVelocity(velocity)
```

Comment:
Maybe broadcast this to all viewers regardless of whether the player has a session or not? This would resolve the jittery knockback problem